xbmc/addons/Repository.cpp

```diff
@@ -55,6 +56,35 @@ using namespace KODI::MESSAGING;
 
 using KODI::MESSAGING::HELPERS::DialogResponse;
 
+CRepository::ResolveResult CRepository::ResolvePathAndHash(const AddonPtr& addon) const
+{
+  CCurlFile file;
+  CURL url{addon->Path()};
+  // Do not follow mirror redirect, we want the headers of the redirect response
+  url.SetProtocolOption("redirect-limit", "0");
+  if (!file.Open(url))
+  {
+    CLog::Log(LOGWARNING, "Could not fetch addon location and hash from {}", url.Get());
+    return {};
+  }
+
+  // Return the location from the header so we don't have to look it up again
+  // (saves one request per addon install)
+  std::string location = file.GetHttpHeader().GetValue("location");
```

Comment:
better use https://curl.haxx.se/libcurl/c/CURLINFO_REDIRECT_URL.html
and maybe also https://curl.haxx.se/libcurl/c/CURLOPT_HEADERFUNCTION.html

Reply:
I'd be possible to get all headers of all requests (incl 30x) while directly following the redirect.

Reply:
We do not want to do that, at least with the current logic. We first get the hash, check if we have the file with that hash in the package cache already, and only if not download the file.